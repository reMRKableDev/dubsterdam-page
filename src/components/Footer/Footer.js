import React from "react";
import Emoji from "../Emoji/Emoji";
import {
  StyledFooter,
  StyledH2,
  StyledFooterPhrase,
  StyledFooterDeveloperTag,
} from "./FooterStyles";

const Footer = () => {
  return (
    <StyledFooter>
      <StyledH2>
        <span>Thank You</span>
      </StyledH2>
      <StyledFooterPhrase>for checking us out!</StyledFooterPhrase>
      <StyledFooterDeveloperTag>
        Made with <Emoji symbol="❤️" label="love" /> by{" "}
        <a href="https://remrkabledev.com/">reMRKable Dev</a>
      </StyledFooterDeveloperTag>
    </StyledFooter>
  );
};

export default Footer;
