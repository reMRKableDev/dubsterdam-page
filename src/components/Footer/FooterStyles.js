import styled from "styled-components";
import { footerColor, brandColor, bgColor } from "../../utils/themeColors";
import { respondTo } from "../../layout/respondTo";

export const StyledFooter = styled.footer`
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 300px;
  background: ${footerColor};
  z-index: -1;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const StyledH2 = styled.h2`
  font-size: 6vw;
  font-family: "Rubik", sans-serif;
  margin: 1em 0;
  text-transform: uppercase;

  span {
    color: ${brandColor};
    border: 4px solid ${brandColor};
    box-shadow: 0 0 10px ${!bgColor};
    padding: 5px;
    text-transform: none;
  }

  ${respondTo.sm`
    font-size: 5.5vw;
    margin: 0.8em 0;

    span {
        border: 6px solid ${brandColor};
    }
  `}

  ${respondTo.md`
    font-size: 5vw;
    margin: 0.6em 0;

    span {
        border: 8px solid ${brandColor};
    }
  `}
  
  ${respondTo.lg`
    font-size: 4.5vw;
    margin: 0.5em 0;

    span {
        border: 10px solid ${brandColor};
    }
  `}
`;

export const StyledFooterPhrase = styled.p`
  font-family: "Karla", sans-serif;
  color: ${bgColor};
  margin-bottom: 25px;
`;

export const StyledFooterDeveloperTag = styled.p`
  font-family: "Rubik", sans-serif;
  color: ${bgColor};

  a {
    color: ${brandColor};
    text-decoration: none;
    cursor: pointer;
  }
`;
