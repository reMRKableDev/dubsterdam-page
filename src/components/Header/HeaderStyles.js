import styled from "styled-components";
import { bgColor, brandColor } from "../../utils/themeColors";
import { respondTo } from "../../layout/respondTo";
import { fadeInUpAnimation, shakeAnimation } from "../../utils/animations";

export const StyledHeader = styled.header`
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
`;

export const StyledFigure = styled.figure`
  animation: 1.2s ${shakeAnimation};
  animation-delay: 1.4s;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;

  img {
    width: 30vw;
    height: auto;
  }
`;

export const StyledH1 = styled.h1`
  animation: 1.5s ${fadeInUpAnimation};
  font-size: 8vw;
  font-family: "Rubik", sans-serif;
  margin: 1em 0;
  text-transform: uppercase;

  span {
    color: ${brandColor};
    border: 4px solid ${brandColor};
    box-shadow: 0 0 10px ${!bgColor};
    padding: 5px;
    text-transform: none;
  }

  ${respondTo.sm`
    margin: 0.8em 0;

    span {
      border: 6px solid ${brandColor};
    }
  `}

  ${respondTo.md`
  margin: 0.6em 0;

  span {
    border: 8px solid ${brandColor};
  }
  `}
  
  ${respondTo.lg`
  margin: 0.5em 0;

  span {
    border: 10px solid ${brandColor};
  }
	`}
`;
