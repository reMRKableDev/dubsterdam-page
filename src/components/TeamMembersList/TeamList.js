import React from "react";
import { StudentList } from "../../data/students";
import Member from "../TeamMember/Member";
import { StyledH2, StyledSubPhrase, StyledTeamDiv } from "./TeamListStyles";

const TeamList = () => {
  return (
    <section>
      <StyledH2>The Team</StyledH2>
      <StyledSubPhrase>
        Switch theme-modes and we will greet you in a new language!
      </StyledSubPhrase>
      <StyledTeamDiv>
        {StudentList.map((student) => (
          <Member key={student.id} student={student} />
        ))}
      </StyledTeamDiv>
    </section>
  );
};

export default TeamList;
