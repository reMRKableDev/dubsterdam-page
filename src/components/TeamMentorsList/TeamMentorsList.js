import React from "react";
import { MentorsList } from "../../data/mentors";
import Member from "../TeamMember/Member";
import { StyledH2, StyledTeamDiv } from "./TeamMentorsListStyles";

const TeamMentorsList = () => {
  return (
    <section>
      <StyledH2>The Mentors</StyledH2>
      <StyledTeamDiv>
        {MentorsList.map((mentor) => (
          <Member key={mentor.id} mentor={mentor} />
        ))}
      </StyledTeamDiv>
    </section>
  );
};

export default TeamMentorsList;
