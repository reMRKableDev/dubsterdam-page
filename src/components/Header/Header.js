import React from "react";
import wcsLogo from "../../assets/logo.png";

import { StyledHeader, StyledFigure, StyledH1 } from "./HeaderStyles";

const Header = () => {
  return (
    <StyledHeader>
      <StyledFigure>
        <img src={wcsLogo} alt="logo" />
      </StyledFigure>
      <StyledH1>
        We are <span>DubSterdam</span>
      </StyledH1>
    </StyledHeader>
  );
};

export default Header;
