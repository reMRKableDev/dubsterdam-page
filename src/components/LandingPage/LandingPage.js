import React from "react";

import Header from "../Header/Header";
import TeamList from "../TeamMembersList/TeamList";
import TeamMentorsList from "../TeamMentorsList/TeamMentorsList";
import Footer from "../Footer/Footer";

const LandingPage = ({ theme, handleThemeToggle }) => {
  return (
    <div>
      <Header />
      <TeamList />
      <TeamMentorsList />
      <Footer />
    </div>
  );
};

export default LandingPage;
