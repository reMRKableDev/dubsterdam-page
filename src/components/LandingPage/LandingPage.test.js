import React from "react";
import { shallow } from "enzyme";

import LandingPage from "./LandingPage";
import Header from "../Header/Header";
import TeamList from "../TeamMembersList/TeamList";
import TeamMentorsList from "../TeamMentorsList/TeamMentorsList";
import Footer from "../Footer/Footer";

import {
  validateMatchSnapshot,
  validateToEqual,
} from "../../utils/testValidators";

describe("Landing Page unit tests", () => {
  it("should validate match snapshot for Landing Page component ", () => {
    const wrapper = shallow(<LandingPage />);
    validateMatchSnapshot(wrapper);
  });

  it("should validate presence of child component Header ", () => {
    const wrapper = shallow(<LandingPage />);
    const childComponent = wrapper.find(Header).length;

    validateToEqual(childComponent, 1);
  });

  it("should validate presence of child component TeamList ", () => {
    const wrapper = shallow(<LandingPage />);
    const childComponent = wrapper.find(TeamList).length;

    validateToEqual(childComponent, 1);
  });

  it("should validate presence of child component TeamMentorsList ", () => {
    const wrapper = shallow(<LandingPage />);
    const childComponent = wrapper.find(TeamMentorsList).length;

    validateToEqual(childComponent, 1);
  });

  it("should validate presence of child component Footer ", () => {
    const wrapper = shallow(<LandingPage />);
    const childComponent = wrapper.find(Footer).length;

    validateToEqual(childComponent, 1);
  });
});
