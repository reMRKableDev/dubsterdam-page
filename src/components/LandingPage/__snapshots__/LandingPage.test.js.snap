// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page unit tests should validate match snapshot for Landing Page component  1`] = `
<div>
  <Header />
  <TeamList />
  <TeamMentorsList />
  <Footer />
</div>
`;
